tests/sstable_3_x_test.cc

```diff
@@ -1741,6 +1741,52 @@ SEASTAR_THREAD_TEST_CASE(test_compressed_partition_key_with_values_of_different_
         COMPRESSED_PARTITION_KEY_WITH_VALUES_OF_DIFFERENT_TYPES_PATH, compressor::lz4);
 }
 
+// Following test runs on files in tests/sstables/3.x/compressed/zstd_multiple_chunks.
+// Size of data in the sstables is big enough that multiple compressed chunks were used.
+// The files were created using following CQL statements:
+//
+// CREATE KEYSPACE test_ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
+//
+// CREATE TABLE test_ks.test_table (val1 INT, val2 INT, PRIMARY KEY (val1, val2))
+//      WITH compression = {'sstable_compression': 'org.apache.cassandra.io.compress.ZstdCompressor',
+//                          'compression_level': 5,
+//                          'chunk_length_in_kb': 4};
+//
+// And for each `i` in the range [1, 2000],
+// INSERT INTO test_ks.test_table (val1, val2) VALUES (0, i);
+
+static thread_local const sstring ZSTD_MULTIPLE_CHUNKS_PATH =
+    "tests/sstables/3.x/compressed/zstd_multiple_chunks";
+static thread_local const schema_ptr ZSTD_MULTIPLE_CHUNKS_SCHEMA =
```

Comment:
Please no more `thread_local`. We want to get rid of these.

Reply:
I removed thread_local from the path, but I can't remove it from the schema builder, because I get errors like this one:
```
unknown location(0): fatal error: in "test_uncompressed_partition_key_with_values_of_different_types_read": sstables::malformed_sstable_exception: bool_val definition in serialization header does not match schema. Expected org.apache.cassandra.db.marshal.BooleanType but got org.apa
che.cassandra.db.marshal.BooleanType in sstable tests/sstables/3.x/uncompressed/partition_key_with_values_of_different_types/mc-1-big-Data.db                                                                                                                                             
```
most probably the problem is that these datatype singletons are thread local. Comparing the type is implemented by comparing the addresses, which are different. There should be a separate issue and pull request that fixes that.
Or should there? It's good that these singletons are thread local. We don't want shards accessing memory of other shards too often.

Reply:
The real solution would be to move the schema building into a function, and have each test build it, instead of using a shared instance. This is perpendicular to this issue, so you can leave it as-is, but let's not make the problem worse.

Reply:
Those global thread_locals (the problem isn't that they're thread_local, but global) are not too bad in leaf files like unit tests. They cause harm when they're in infrastructure, where you can't use one thing without the other because everything depends indirectly on every other thing in a tangled mess.